### PR TITLE
Review Swift codes

### PR DESCRIPTION
**Pull request by nuomi1**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [x] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

**Comment by nuomi1** [comment]
stride by 1

1. `stride(from: x, to: y, by: 1)` to `x ..< y`
2. `stride(from: x, through: y, by: 1)` to `x ... y`
3. `stride(from: 0, to: nums.count, by: 1)` to `nums.indices`
4. `stride(from: x, to: nums.count, by: 1)` to `nums.indices.dropFirst(x)`

---

stride by -1

1. `stride(from: n, to: 0, by: -1)` to `(1 ... n).reversed()`
2. `stride(from: n, through: 0, by: -1)` to `(0 ... n).reversed()`
3. `stride(from: nums.count - 1, to: 0, by: -1)` to `nums.indices.dropFirst().reversed()`
4. `stride(from: nums.count - 1, through: 0, by: -1)` to `nums.indices.reversed()`

---

sequence

1. `sequence(first: y - 1, next: { $0 >= x + 1 ? $0 - 1 : nil })` to `(x ..< y).reversed()`
2. `sequence(first: nums.count - 1, next: { $0 >= 0 + 1 ? $0 - 1 : nil })` to `nums.indices.reversed()`

---

index

1. `0` to `nums.startIndex` and `nums.count - 1` to `nums.endIndex - 1`

---

other

1. set default value in `init()`
2. use `func xxx()` instead of `property _xxx`
4. use `map()` and `compactMap()`
5. use `swapAt()`
6. use `??`
7. format big number

**Comment by nuomi1** [comment]
@rakuyoMo review pls.

File: codes/swift/chapter_heap/heap.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    PrintUtil.printHeap(queue: heap.unordered)
+}
+
+@main
+enum _Heap {
+    /* Driver Code */
+    static func main() {
+        /* 初始化堆 */
+        // Swift 的 Heap 类型同时支持最大堆和最小堆
+        var heap = Heap<Int>()
```
Comment (krahets) [inline codes/swift/chapter_heap/heap.swift:28]: We should also add the code to `heap.md` referring to other programming languages. You can add it through another PR.

**Comment by rakuyoMo** [comment]
You are really careful and thought of using `startIndex` and `endIndex` to enhance the robustness of the code when it comes to indexing, which is really great!

**Comment by krahets** [comment]
Thanks for the review @rakuyoMo. Could we merge this PR @nuomi1?

**Comment by rakuyoMo** [comment]
> stride by 1
> 
> 1. `stride(from: x, to: y, by: 1)` to `x ..< y`
> 2. `stride(from: x, through: y, by: 1)` to `x ... y`
> 3. `stride(from: 0, to: nums.count, by: 1)` to `nums.indices`
> 4. `stride(from: x, to: nums.count, by: 1)` to `nums.indices.dropFirst(x)`
> 
> stride by -1
> 
> 1. `stride(from: n, to: 0, by: -1)` to `(1 ... n).reversed()`
> 2. `stride(from: n, through: 0, by: -1)` to `(0 ... n).reversed()`
> 3. `stride(from: nums.count - 1, to: 0, by: -1)` to `nums.indices.dropFirst().reversed()`
> 4. `stride(from: nums.count - 1, through: 0, by: -1)` to `nums.indices.reversed()`

Why is the `dropFirst` method called on the `indices` property? I feel that this will bring additional performance overhead. It is better to directly construct a `Range` object like the first and second items.

**Review by rakuyoMo — COMMENTED** [review]
I'm done reviewing. There are some small ideas, but they don't hinder the incorporation. If @nuomi1 is not prepared to modify them, then I will approve this pull.

File: codes/swift/chapter_graph/graph_adjacency_list.swift [hunk 2]
```diff
@@ -72,12 +72,9 @@ public class GraphAdjList {
...
     public func print() {
         Swift.print("邻接表 =")
-        for pair in adjList {
-            var tmp: [Int] = []
-            for vertex in pair.value {
-                tmp.append(vertex.val)
-            }
-            Swift.print("\(pair.key.val): \(tmp),")
+        for (vertex, list) in adjList {
+            let list = list.map { $0.val }
```
[the remark this answers is not in the document]
Comment (nuomi1) [reply codes/swift/chapter_graph/graph_adjacency_list.swift:76]: Always use trailing closures.

**Comment by nuomi1** [comment]
> > stride by 1
> > 
> > 1. `stride(from: x, to: y, by: 1)` to `x ..< y`
> > 2. `stride(from: x, through: y, by: 1)` to `x ... y`
> > 3. `stride(from: 0, to: nums.count, by: 1)` to `nums.indices`
> > 4. `stride(from: x, to: nums.count, by: 1)` to `nums.indices.dropFirst(x)`
> > 
> > stride by -1
> > 
> > 1. `stride(from: n, to: 0, by: -1)` to `(1 ... n).reversed()`
> > 2. `stride(from: n, through: 0, by: -1)` to `(0 ... n).reversed()`
> > 3. `stride(from: nums.count - 1, to: 0, by: -1)` to `nums.indices.dropFirst().reversed()`
> > 4. `stride(from: nums.count - 1, through: 0, by: -1)` to `nums.indices.reversed()`
> 
> Why is the `dropFirst` method called on the `indices` property? I feel that this will bring additional performance overhead. It is better to directly construct a `Range` object like the first and second items.
> 
> For example, change `nums.indices.dropFirst(x)` to `x ..< nums.endIndex`.

I think we should handle the stepping of index from the `indices` property of the `Collection` type. Because index is not always of `Int` type.

File: codes/swift/chapter_stack_and_queue/linkedlist_deque.swift [hunk 3]
```diff
@@ -113,13 +113,19 @@ class LinkedListDeque {
     }
 
     /* 访问队首元素 */
-    func peekFirst() -> Int? {
-        isEmpty() ? nil : front?.val
+    func peekFirst() -> Int {
+        if isEmpty() {
```
[the remark this answers is not in the document]
Comment (krahets) [reply codes/swift/chapter_stack_and_queue/linkedlist_deque.swift:117]: I think both are OK and `isEmpty()` is more semantic. (When the queue is empty, return nil; when the queue contains elements(not empty), return the front.val)

[the remark this answers is not in the document]
Comment (rakuyoMo) [reply codes/swift/chapter_stack_and_queue/linkedlist_deque.swift:117]: We need to ensure that `front` has a value when the queue is not empty.Semantically this is unquestionable, but in the code implementation it does require additional coding to ensure this.

But there's nothing wrong with using `isEmpty` from a functionality standpoint; it does what's needed well.

**Comment by rakuyoMo** [comment]
> I think we should handle the stepping of index from the `indices` property of the `Collection` type. Because index is not always of `Int` type.

Indeed.You're right.

----

I'll authorize a review later.

**Review by rakuyoMo — APPROVED** [review]

**Comment by rakuyoMo** [comment]
@krahets I have completed the review.

**Review by krahets — APPROVED** [review]
Thanks @nuomi1 @rakuyoMo !

**Pull request merged**